### ISSUE TITLE
Feature: rewrite dict.get(key, None) as dict.get(key)

### ISSUE THREAD

**Issue by janosh** [charliermarsh/ruff#3546]
This is not in [`flake8-simplify`](https://github.com/MartinThoma/flake8-simplify) AFAIK but would prob best fit in that rule class.

From the flake8-simplify readme:

> Simplifying usage of dictionaries:
> 
> https://github.com/MartinThoma/flake8-simplify/issues/72: Use 'a_dict.get(key, "default_value")' instead of an if-block ([example](https://github.com/MartinThoma/flake8-simplify#SIM401))
> https://github.com/MartinThoma/flake8-simplify/issues/40: Use 'key in dict' instead of 'key in dict.keys()' ([example](https://github.com/MartinThoma/flake8-simplify#SIM118))

**Comment by charliermarsh** [comment]
Yeah just not sure what code this would fall under...

**Comment by janosh** [comment]
Rule name in `flake8-simplify` [will be SIM910](https://github.com/MartinThoma/flake8-simplify/issues/171#issuecomment-1487693798).

**Comment by kyoto7250** [comment]
I am currently working for making the rule in `flake8-simplify`, and I would also like to implement the rule in `ruff`.

**Comment by Borda** [comment]
> I would also like to implement the rule in `ruff`

cool, was this already added? I see in the docs that it shall be implemented, but with 0.0.262 these cases are not flagged or repaired... could be some typing limitation that is not properly recognized? But `flake8-simplify` detects them...

https://github.com/Lightning-AI/lightning/blob/8dac2512733bae034f437334a71e4c114d458a7c/src/lightning/pytorch/profilers/pytorch.py#L322
```
src/lightning/pytorch/profilers/pytorch.py:322:20: SIM910 Use 'profiler_kwargs.get("schedule")' instead of 'profiler_kwargs.get('schedule', None)'
```

https://github.com/Lightning-AI/lightning/blob/8dac2512733bae034f437334a71e4c114d458a7c/src/lightning/pytorch/utilities/parsing.py#L102
```
src/lightning/pytorch/utilities/parsing.py:102:16: SIM910 Use 'local_vars.get(self_var)' instead of 'local_vars.get(self_var, None)'
```

**Comment by janosh** [comment]
> This is implemented and released. In ruff, this will only work on dict literals ({}) while in flake8-simplify this works with any call which resembles the expression (<name>.get("...", None))

Ah, I was wondering if that was the case. I think mentioning in the docs would be great.

**Comment by kyoto7250** [comment]
I implemented this rule, but this difference in behavior was not intended.

@charliermarsh 

I want to fix this behavior.
Is there any way to find out the type of a variable when checking ruff?
